modules/solid_mechanics/src/materials/ADComputeIncrementalShellStrain.C

```diff
@@ -260,6 +280,18 @@ ADComputeIncrementalShellStrain::computeProperties()
                                       _unrotated_total_strain *
                                       (*_contravariant_transformation_matrix[j])[i].transpose();
     }
+    // the element's local coordinate is saved here for the post processing and visualization
```

Comment:
I'm not sure this is a great idea to do this every time this gets called. A few alternatives to think about:
- Do this optionally, only if the user asks for it
- Make these direction vectors stateful properties, so you're just computing them once at the beginning like we're doing for the B matrix, etc. This would require a little memory but would cut down on compute cost.
- Have an AuxKernel that computes this

Maybe we should test how much overhead is involved in this. If it's negligible, then just keep it as-is.

Reply:
Agreed, I added an AuxKernel (ShellLocalCoordinatesAux) that saves the vector components in Aux variables.